### PR TITLE
Feature/2549 add missing slot apis for channels

### LINKED ISSUE #2549: [Compose] - Add missing Slot APIs to Channels features
### Context

We need more customization options for our Channels features.

### Change Proposal

Add slot APIs to the Channelnfo component, that represents the following slots:
- Header/channel title header
- Participants/Members section
- Channel options

Make sure these are decoupled and public components by default and that our users can customize these to their required extent.

Add slot APIs to the DefaultChannelItem component, that represents the following slots:
- Avatar/Leading content that shows the image by default
- Channel details content that shows the name and the last message preview
- Trailing content that shows a timestamp at the end, as well as the unread count and/or the message read state

Also provide various misc slots:
- Channel title slot
- Last message preview slot
- Last message timestamp slot

These slots would be a part of the Details content and Trailing content slot and can be individually replaced if needed

### Acceptance criteria

To accept this task we need to complete the following items:

- [ ] Expose existing component sections as public API that can be customized and reused by our users
- [ ] Introduce new slots as defined above
- [ ] Apply default customization options for these slots, but allow for overriding
- [ ] Test to make sure everything works and can be customized

### PR DESCRIPTION
**Pull request by filbabic**
Fixes #2549 

### 🎯 Goal

We need to support more dynamic customization options for our channel items, in the form of Slot APIs.

### 🛠 Implementation details

Added more slot APIs to the `DefaultChannelItem`, added more documentation and examples.

Exposing now four Slot APIs for the Channel item:
- Leading content
- Details content
- Trailing content
- Divider

Also updated the ChannelActivity examples and the documentation.

### 🎨 UI Changes

We can now customize the UI to achieve following behavior, without replacing the `DefaultChannelItem`. We can obviously customize it to more extent, but this can serve as a good and simple example.

compose_customized_default_channel_item

### ☑️ Checklist

- [x] PR targets the `develop` branch
- [x] Changelog is updated with client-facing changes
~~- [ ] New code is covered by unit tests~~
- [x] Comparison screenshots added for visual changes
- [x] Affected documentation updated (docusaurus, tutorial, CMS (task created))
- [x] Reviewers added

### 🎉 GIF

conversation

### CODE REVIEW

File: docusaurus/docs/Android/04-compose/03-channel-components/04-default-channel-item.mdx [hunk 1]
```diff
@@ -0,0 +1,203 @@
...
+
+```kotlin
+override fun onCreate(savedInstanceState: Bundle?) {
+    super.onCreate(savedInstanceState)
+
+    setContent {
+        ChatTheme {
+			ChannelList(
+				itemContent = { channel ->
+					DefaultChannelItem(
```
Comment (skydoves) [inline docusaurus/docs/Android/04-compose/03-channel-components/04-default-channel-item.mdx:74]: Indentation is growing up. 😄

File: stream-chat-android-compose-sample/src/main/java/io/getstream/chat/android/compose/sample/ChannelActivity.kt [hunk 2]
```diff
@@ -95,10 +108,45 @@ class ChannelActivity : AppCompatActivity() {
...
+                }
+            )
         }
     }
 
+    /**
+     * An example of a customized DefaultChannelItem component.
+     */
+    @Composable
+    fun CustomChannelListItem(channel: Channel, user: User?) {
```
Comment (skydoves) [inline stream-chat-android-compose-sample/src/main/java/io/getstream/chat/android/compose/sample/ChannelActivity.kt:123]: Small thing: Can be `private` this method?

File: stream-chat-android-compose-sample/src/main/java/io/getstream/chat/android/compose/sample/ChannelActivity.kt [hunk 3]
```diff
@@ -95,10 +108,45 @@ class ChannelActivity : AppCompatActivity() {
...
+                )
+            },
+            trailingContent = {
+                Spacer(modifier = Modifier.width(8.dp))
+            },
+            detailsContent = {
+                Text(
+                    text = it.getDisplayName(),
+                    style = ChatTheme.typography.bodyBold
+                )
```
Comment (skydoves) [inline stream-chat-android-compose-sample/src/main/java/io/getstream/chat/android/compose/sample/ChannelActivity.kt:145]: How about adding the `textColor`? We can't see the message on the dark mode. 😃 
```kotlin
color = ChatTheme.colors.textHighEmphasis
```

**Review by skydoves — APPROVED** [review]
Looks great to me. 👍  I left some small comments.

File: docusaurus/docs/Android/04-compose/03-channel-components/04-default-channel-item.mdx (hunk 1 above)
Comment (filbabic) [reply docusaurus/docs/Android/04-compose/03-channel-components/04-default-channel-item.mdx:74]: Ugh I tried doing them with tabs specifically for this reason, but idk what's going on

File: stream-chat-android-compose-sample/src/main/java/io/getstream/chat/android/compose/sample/ChannelActivity.kt (hunk 3 above)
Comment (filbabic) [reply stream-chat-android-compose-sample/src/main/java/io/getstream/chat/android/compose/sample/ChannelActivity.kt:145]: Good point!

File: docusaurus/docs/Android/04-compose/03-channel-components/04-default-channel-item.mdx [hunk 4]
```diff
@@ -0,0 +1,203 @@
...
+)
+```
+
+* `modifier`: The modifier parameter for the root component. You can apply a background, elevation, padding, shape, touch handlers and much more.
+* `leadingContent`: Customizable composable function that allows you to override the content that appears at the start of the list item. By default, it represents a `ChannelAvatar`.
+* `detailsContent`: Customizable composable function that allows you to override the center part of the list item. By default, it represents the `ChannelDetails` that show its name and the last message preview.
+* `trailingContent`: Customizable composable function that allows you to override the content that appears at the end of the list item. By default, it represents the `ChannelLastMessageInfo` that shows the unread message indicator if there are unread messages and the last message read state.
+* `divider`: Customizable composable function that allows you to override the content that appears at the bottom of the list item, as a divider.
+
+Here's a simple example for building your own channel item, by overriding the `mentioned` parameters:
```
Comment (bychkovdmitry) [inline docusaurus/docs/Android/04-compose/03-channel-components/04-default-channel-item.mdx:147]: Why "mentioned" is surrounded with backticks?

File: docusaurus/docs/Android/04-compose/03-channel-components/04-default-channel-item.mdx [hunk 5]
```diff
@@ -0,0 +1,203 @@
...
+        divider = { // Override the divider
+            Spacer(
+                modifier = Modifier
+                    .fillMaxWidth(0.5f)
+                    .height(0.5.dp)
+                    .align(CenterHorizontally)
+                    .background(color = ChatTheme.colors.textLowEmphasis)
+            )
+        },
+        trailingContent = { // Replace the last message content with a spacer
```
Comment (bychkovdmitry) [inline docusaurus/docs/Android/04-compose/03-channel-components/04-default-channel-item.mdx:182]: Last message content is a bit misleading name, as the last message preview is actually in the `ChannelDetails` component.

File: docusaurus/docs/Android/04-compose/03-channel-components/04-default-channel-item.mdx (hunk 4 above)
Comment (filbabic) [reply docusaurus/docs/Android/04-compose/03-channel-components/04-default-channel-item.mdx:147]: I think I was going to mention a specific param and then accidentally put this in

File: docusaurus/docs/Android/04-compose/03-channel-components/04-default-channel-item.mdx (hunk 5 above)
Comment (filbabic) [reply docusaurus/docs/Android/04-compose/03-channel-components/04-default-channel-item.mdx:182]: Yeah that's good, I'll update to "trailing content"

**Review by bychkovdmitry — APPROVED** [review]
Awesome job!

**Pull request merged**